migrator/migrations/m_109_to_m_110_replace_deprecated_resources/migration.go

```diff
@@ -0,0 +1,134 @@
+package m109tom110
+
+import (
+	"github.com/gogo/protobuf/proto"
+	"github.com/pkg/errors"
+	"github.com/stackrox/rox/generated/storage"
+	"github.com/stackrox/rox/migrator/migrations"
+	"github.com/stackrox/rox/migrator/types"
+	"github.com/tecbot/gorocksdb"
+)
+
+const (
+	batchSize = 500
+)
+
+// Replacement resources
+const (
+	Access              = "Access"
+	DeploymentExtension = "DeploymentExtension"
+	Image               = "Image"
+	Integration         = "Integration"
+)
+
+// Replaced resources
+const (
+	APIToken             = "APIToken"
+	AuthProvider         = "AuthProvider"
+	BackupPlugins        = "BackupPlugins"
+	Group                = "Group"
+	ImageComponent       = "ImageComponent"
+	ImageIntegration     = "ImageIntegration"
+	Indicator            = "Indicator"
+	Licenses             = "Licenses"
+	NetworkBaseline      = "NetworkBaseline"
+	Notifier             = "Notifier"
+	ProcessWhitelist     = "ProcessWhitelist"
+	Risk                 = "Risk"
+	SignatureIntegration = "SignatureIntegration"
+	User                 = "User"
+)
+
+var (
+	migration = types.Migration{
+		StartingSeqNum: 109,
+		VersionAfter:   storage.Version{SeqNum: 110},
+		Run: func(databases *types.Databases) error {
+			return migatePermissionSets(databases.RocksDB)
+		},
+	}
+
+	prefix = []byte("permission_sets")
+
+	replacements = map[string]string{
+		APIToken:             Integration,
+		AuthProvider:         Access,
+		BackupPlugins:        Integration,
+		Group:                Access,
+		ImageComponent:       Image,
+		ImageIntegration:     Integration,
+		Indicator:            DeploymentExtension,
+		Licenses:             Access,
+		NetworkBaseline:      DeploymentExtension,
+		Notifier:             Integration,
+		ProcessWhitelist:     DeploymentExtension,
+		Risk:                 DeploymentExtension,
+		SignatureIntegration: Integration,
+		User:                 Access,
+	}
+
+	readOpts  = gorocksdb.NewDefaultReadOptions()
+	writeOpts = gorocksdb.NewDefaultWriteOptions()
+)
+
+func init() {
+	migrations.MustRegisterMigration(migration)
+}
+
+func propagatePermission(resource string, accessLevel storage.Access, permissions map[string]storage.Access) storage.Access {
+	if _, found := permissions[resource]; !found {
+		return accessLevel
+	}
+	oldLevel := permissions[resource]
+	if accessLevel > oldLevel {
+		return oldLevel
+	}
+	return accessLevel
+}
+
+func migatePermissionSets(db *gorocksdb.DB) error {
```

Comment:
nit: can we be a bit more explicit regarding _what exactly_ we are migrating here? Something such as `migrateReplacedResources`.